fix server build error

```
% npm run build

> server@1.0.0 build
> tsc

src/routes/auth.ts:47:49 - error TS2769: No overload matches this call.
  Overload 1 of 3, '(payload: string | object | Buffer, secretOrPrivateKey: Secret, options?: SignOptions | undefined): string', gave the following error.
    Argument of type 'string | undefined' is not assignable to parameter of type 'Secret'.
      Type 'undefined' is not assignable to type 'Secret'.
  Overload 2 of 3, '(payload: string | object | Buffer, secretOrPrivateKey: Secret, callback: SignCallback): void', gave the following error.
    Argument of type 'string | undefined' is not assignable to parameter of type 'Secret'.

47     const token = jwt.sign({ userId: user.id }, process.env.JWT_SECRET, { expiresIn: '1h' });
                                                   ~~~~~~~~~~~~~~~~~~~~~~


src/routes/chat.ts:68:56 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'number'.

68     const chatThread = await ChatThread.findThreadByPk(threadId);
                                                          ~~~~~~~~


Found 2 errors in 2 files.

Errors  Files
     1  src/routes/auth.ts:47
     1  src/routes/chat.ts:68

```

--- a/server/src/routes/auth.ts
+++ b/server/src/routes/auth.ts
@@ -44,6 +44,10 @@ router.post('/login', async (req, res) => {
       return res.status(401).json({ message: 'Invalid password' });
     }
 
+    if (!process.env.JWT_SECRET) {
+      throw new Error('JWT_SECRET is not defined');
+    }
+
     const token = jwt.sign({ userId: user.id }, process.env.JWT_SECRET, { expiresIn: '1h' });
 
     res.status(200).json({ message: 'Login successful', token });

--- a/server/src/routes/chat.ts
+++ b/server/src/routes/chat.ts
@@ -33,7 +33,7 @@ router.post('/continue', async (req, res) => {
   }
 
   try {
-    const chatThread = await ChatThread.findThreadByPk(threadId);
+    const chatThread = await ChatThread.findThreadByPk(Number(threadId));
 
     if (!chatThread) {
       return res.status(404).json({ message: 'Chat thread not found' });
@@ -65,7 +65,7 @@ router.delete('/delete/:threadId', async (req, res) => {
   const { threadId } = req.params;
 
   try {
-    const chatThread = await ChatThread.findThreadByPk(threadId);
+    const chatThread = await ChatThread.findThreadByPk(Number(threadId));
 
     if (!chatThread) {
       return res.status(404).json({ message: 'Chat thread not found' });

